annotation/tests/test_jobs.py

```diff
@@ -343,4 +328,203 @@ def test_get_job_attributes_for_post_attribute_not_found():
     mock_session = MagicMock()
     mock_session.query().filter_by().first.return_value = None
     with pytest.raises(FieldConstraintError):
-        get_job_attributes_for_post(mock_session, job_id, "test", (Job.name,))
+        annotation.jobs.services.get_job_attributes_for_post(
+            mock_session, job_id, "test", (Job.name,)
+        )
+
+
+def test_recalculate_file_pages(files: Tuple[File]):
+    mock_session = MagicMock()
+    mock_result = MagicMock()
+    mock_session.query().filter.return_value = mock_result
+    expected_validating_pages = [5, 7]
+    expected_annotating_pages = [3, 5, 6, 7]
+
+    filter_side_effect = (
+        lambda *args: MagicMock(
+            all=MagicMock(return_value=[([5, 6, 7],), ([3, 6],)])
+        )
+        if "NOT" in str(args[0])
+        else MagicMock(all=MagicMock(return_value=[([5, 7],)]))
+    )
+    mock_result.filter.side_effect = filter_side_effect
+    annotation.jobs.services.recalculate_file_pages(mock_session, files[0])
+    assert files[0].distributed_validating_pages == expected_validating_pages
+    assert files[0].distributed_annotating_pages == expected_annotating_pages
+
+
+def test_read_user(users: Tuple[User]):
+    mock_session = MagicMock()
+    mock_session.query().get.return_value = users[0]
+    result = annotation.jobs.services.read_user(mock_session, users[0])
+    assert result == users[0]
+
+
+def test_create_user():
+    mock_session = MagicMock()
+    expected_result = User(user_id=1)
+    result = annotation.jobs.services.create_user(mock_session, 1)
+    assert result == expected_result
+
+
+# can not create Job with JobSchema
```

Comment:
move this into the skip reason and add a todo to the problematic function with brief desc of problem, please